### PR TITLE
Remove most inline scripts & Fix appear on scroll - #63

### PR DESCRIPTION
**Pull request by migueldemoura**
Rolled my own appear on scroll script, was getting pretty pissed with the other scripts - should work now.
The other commit moves some inline code to an external js file to improve caching and facilitate CSP rules.

### CODE REVIEW

File: _assets/stylesheets/modules/_scrollappear.scss [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+.appear {
+  visibility: hidden;
+  opacity: 0;
+}
+
+.appeared {
+  visibility: visible;
+  opacity: 1;
+  -webkit-transition: opacity 0.5s ease-in-out;
```
Comment (nielsenramon) [inline _assets/stylesheets/modules/_scrollappear.scss]: We use autoprefixer, so you can remove the vendor prefixes :-)

File: _config.yml [hunk 2]
```diff
@@ -11,6 +11,7 @@ about_enabled: false # Change to true if you wish to show an icon in the navigat
 discus_identifier: # Add your Disqus identifier
 ga_analytics: # Add your GA Tracking Id
 local_fonts: false # Change to true if you wish to use local fonts
+scrollappear_enabled: true # Change to false to disable
```
Comment (nielsenramon) [inline _config.yml]: I would put this under rss_enabled just to work alphabetically.

**Comment by nielsenramon** [comment]
@migueldemoura Looking great already, I am having the same issues as scrollreveal now tho. I get white pages randomly in Chrome. Refreshing works. Maybe Turbolinks is messing with this after all.

**Comment by migueldemoura** [comment]
@nielsenramon, I can't reproduce the issue anymore, seems to be working here on Chromium. Can you see if the turbolinks event gets triggered?

**Comment by nielsenramon** [comment]
@migueldemoura I'm ok dropping it, we aren't really getting that much benefit out of it anyway.

**Comment by migueldemoura** [comment]
@nielsenramon, let me know if you want the appear on scroll to also apply to the elements of the article itself :smile:

File: _assets/javascripts/application.js [hunk 3]
```diff
@@ -1,30 +1,17 @@
 //= require_self
 
-// Share buttons
-$('.btn-share').on('click', function() {
-  window.open($(this).attr('href'), 'Share', 'width=200,height=200,noopener');
-  return false;
-});
+$(document).ready(function() {
```
Comment (nielsenramon) [inline _assets/javascripts/application.js]: Let's use the jQuery short notation here :-)

File: _includes/head.html [hunk 4]
```diff
@@ -31,12 +30,12 @@
   <link href="{{ 'feed.xml' | absolute_url }}" type="application/rss+xml" rel="alternate" title="{{ site.name }} Last 10 blog posts" />
 
   {% if site.local_fonts %}
-    {% stylesheet fonts %}
+    {% include stylesheet.html path="fonts" %}
   {% endif %}
 
   {% if site.blog_theme == "light" %}
-    {% stylesheet light %}
+    {% include stylesheet.html path="light" %}
```
Comment (nielsenramon) [inline _includes/head.html]: Why is include required here?

**Comment by nielsenramon** [comment]
@migueldemoura Nah that's fine 😄  Great PR. Commented some small things!

File: _includes/head.html (hunk 4 above)
Comment (migueldemoura) [inline _includes/head.html]: Only reason is that jekyll-assets outputs unnecessary tags for html5 `</script>` and `<link>`:

`<link type="text/css" rel="stylesheet" href="/assets/light.css">`
,should be:
`<link rel="stylesheet" href="/assets/light.css">`

and

`<script type="text/javascript" src="/assets/vendor.js"></script>`
,should be:
`<script src="/assets/vendor.js"></script>`

**Comment by nielsenramon** [comment]
@migueldemoura Very nice work, can't thank you enough!

**Comment by migueldemoura** [comment]
@nielsenramon, thank you for the great theme :smile:

**Pull request merged**